Develop

garcijo opened the pull request:
Resolved #4:restructure project folders, implemented unit tests for Web/Domain, and corrected some styling errors.

garcijo committed:
Added UserMapper unit tests and changed password formats

Created a unit test file for UserMapper. Also changed the users password
encription from MySQL's password to md5.

garcijo committed:
Added tests for SpotifyFeed

Created a unit test file for SpotifyFeed. Also did some minor changes to
schema.sql so that when a user is removed all their liked songs are also
deleted.

garcijo committed:
Readme updated and minor style fixes.

garcijo committed:
Added some missing javadoc to various functions and classes

garcijo committed:
Changed database object from PDO to Database

Changed the $db variable type in all the tests since all classes in
Domain use Database and not PDO.

adamaltman commented on src/Web/Action/LoginAction.php:
```diff
@@ -28,7 +31,7 @@ public function __invoke(Request $request, Response $response, $args)
         $data = $request->getParsedBody();
         $userEmail = filter_var($data['email'], FILTER_SANITIZE_STRING);
-        $userPass = filter_var($data['password'], FILTER_SANITIZE_STRING);
+        $userPass = md5(filter_var($data['password'], FILTER_SANITIZE_STRING));
```
`md5` maybe for a demo app, but not a proper way to handle password authentication in 2016.

adamaltman commented on src/Web/Action/SignupAction.php:
```diff
@@ -29,7 +32,7 @@ public function __invoke(Request $request, Response $response, $args)
         $userName = filter_var($data['name'], FILTER_SANITIZE_STRING);
         $userEmail = filter_var($data['email'], FILTER_SANITIZE_STRING);
-        $userPass = filter_var($data['password'], FILTER_SANITIZE_STRING);
+        $userPass = md5(filter_var($data['password'], FILTER_SANITIZE_STRING));
```
I'll make a ticket about this...

adamaltman commented on src/Web/Domain/SpotifyFeed.php:
```diff
@@ -90,40 +96,41 @@ public function getMusic(string $user):array
+        $songImg = $song->album->images[1]->url;
+        $_POST['song_img'] = $songImg;
+        $songWidth = $song->album->images[1]->width;
+        $_POST['song_width'] = $songWidth;
```
confused by why you are setting global values here?

adamaltman commented on tests/Web/Domain/UserMapperTest.php:
```diff
@@ -0,0 +1,100 @@
+        //remove dummy data
+        $userMapper->removeUser($testUserName);
+
+        return $testLog;
```
why `return` anything?

garcijo committed:
Changed password verification methods

Instead of using md5, the website now uses password_hash and
password_verify

garcijo commented:
resolved #12: php password. As for the global POST data, that is being read in the home page to render the music player. The code for that is in templates/home.phtml

garcijo committed:
removed post variables from home.phtml

garcijo committed:
Smoothened player loading

Added a blank image and changed the fading values in order to create a
smoother look for when the player loads.

garcijo committed:
PHP style fix
